aspnetcore/blazor/hosting-models.md

```diff
@@ -34,65 +59,77 @@ Blazor is a web framework for building web UI components ([Razor components](xre
+On the client, the Blazor script establishes the SignalR connection with the server. The script is served from an embedded resource in the ASP.NET Core shared framework.
 
 The Blazor Server hosting model offers several benefits:
 
-* Download size is significantly smaller than a Blazor WebAssembly app, and the app loads much faster.
+* Download size is significantly smaller than when the Blazor WebAssembly hosting model is used, and the app loads much faster.
 * The app takes full advantage of server capabilities, including the use of .NET Core APIs.
 * .NET Core on the server is used to run the app, so existing .NET tooling, such as debugging, works as expected.
-* Thin clients are supported. For example, Blazor Server apps work with browsers that don't support WebAssembly and on resource-constrained devices.
+* Thin clients are supported. For example, Blazor Server works with browsers that don't support WebAssembly and on resource-constrained devices.
 * The app's .NET/C# code base, including the app's component code, isn't served to clients.
 
 The Blazor Server hosting model has the following limitations:
 
 * Higher latency usually exists. Every user interaction involves a network hop.
-* There's no offline support. If the client connection fails, the app stops working.
+* There's no offline support. If the client connection fails, interactivity fails.
 * Scaling apps with many users requires server resources to handle multiple client connections and client state.
 * An ASP.NET Core server is required to serve the app. Serverless deployment scenarios aren't possible, such as serving the app from a Content Delivery Network (CDN).
 
-We recommend using the [Azure SignalR Service](/azure/azure-signalr) for Blazor Server apps. The service allows for scaling up a Blazor Server app to a large number of concurrent SignalR connections.
+We recommend using the [Azure SignalR Service](/azure/azure-signalr) for apps that adopt the Blazor Server hosting model. The service allows for scaling up a Blazor Server app to a large number of concurrent SignalR connections.
 
 ## Blazor WebAssembly
 
-Blazor WebAssembly (WASM) apps run client-side in the browser on a WebAssembly-based .NET runtime. The Blazor app, its dependencies, and the .NET runtime are downloaded to the browser. The app is executed directly on the browser UI thread. UI updates and event handling occur within the same process. The app's assets are deployed as static files to a web server or service capable of serving static content to clients.
+Blazor WebAssembly (WASM) runs client-side in the browser on a WebAssembly-based .NET runtime. Razor components, their dependencies, and the .NET runtime are downloaded to the browser. Components are executed directly on the browser UI thread. UI updates and event handling occur within the same process. Assets are deployed as static files to a web server or service capable of serving static content to clients.
```

Comment:
> Components are executed directly on the browser UI thread.

With the multithreading work there's some discussion about potentially changing this in a future release, so we should remember to update this bit if that eventually happens. The current text is accurate and fine for now.